spec/features/ingredients/index_spec.rb

```diff
@@ -0,0 +1,35 @@
+# User Story 1 - Ingredients Index
+
+# As a visitor,
+# When I visit '/ingredients'
+# I see a list of all the ingredients including their name and cost
+# (e.g. "Ground Beef: 2"
+#      "Salt: 4")
+
+require "rails_helper"
+
+RSpec.describe "User Story #1 - Ingredient Index", type: :feature do
+  describe "When I visit '/ingredients'" do
+    it "I see a list of all the ingredients - including their name and cost" do
+      ingredient_1 = Ingredient.create!(name: "pasta", cost: 12)
+      ingredient_2 = Ingredient.create!(name: "tomatoes", cost: 2)
+      ingredient_3 = Ingredient.create!(name: "pesto", cost: 5)
+      ingredient_4 = Ingredient.create!(name: "cheese", cost: 7)
+
+      visit "/ingredients"
+      save_and_open_page
+
+      # Happy Path - all of these should appear because they are instantiated in our test
+      expect(page).to have_content("pasta")
+      expect(page).to have_content("tomatoes")
+      expect(page).to have_content("pesto")
+      expect(page).to have_content("cheese")
+      
+      # Sad Path - None of these should appear because they were never created
+      expect(page).to_not have_content("olives")
+      expect(page).to_not have_content("meat")
+      expect(page).to_not have_content("bread")
+      expect(page).to_not have_content("wine")
+    end
```

Comment:
Your test is incomplete. Your assertions should be checking for the name as well as the cost of the each ingredient in the view, but right now you are only checking for the name.

Also, your sad paths are not really sad paths and don't add a lot to your tests. You're checking for the page to not have names of ingredients that were never created, so I am not sure how they would even end up in the view in the first place.

Sad paths are not for every single test, and are usually reserved for a possible path that an end user would take, like not entering info into a form, for example.